executor/index_lookup_hash_join.go

```diff
@@ -174,6 +201,51 @@ func (e *IndexNestedLoopHashJoin) Next(ctx context.Context, req *chunk.Chunk) er
 	return nil
 }
 
+func (e *IndexNestedLoopHashJoin) runInOrder(ctx context.Context, req *chunk.Chunk) error {
+	var (
+		result *indexHashJoinResult
+		ok     bool
+	)
+	for {
+		task := e.fetchNextTask(ctx)
```

Comment:
How about changing the returned argument of `fetchNextTask` from `task *Task` to `hasNewTask bool`?
Then we can remove the local variable `task` and always use the `e.curTask` to reference the current task.